### PR TITLE
Add new langNumFmtFunc in `numfmt`

### PR DESCRIPTION
**Pull request by wushiling50**
# PR Details
Support for more built-in langNumFmt allows `GetCellValue` to fetch dates and times in more localizations

## Related Issue
#1885 

## Motivation and Context
When I using the following code, I found that the current `CultureInfo` type only supports `CultureNameZhCN` and `CultureNameEnUS`, which makes it impossible for me to obtain the time and date formats of other regions through the `GetCellValue` method.
```
package main

import (
	"fmt"

	"github.com/xuri/excelize/v2"
)

func main() {
	f := excelize.NewFile(excelize.Options{
		CultureInfo: excelize.CultureNameZhCN, // or excelize.CultureNameEnUS
	})

	style1, err := f.NewStyle(&excelize.Style{
		NumFmt: 27,
	})
	if err != nil {
		fmt.Println(err)
		return
	}
	f.SetCellStyle("Sheet1", "A2", "A2", style1)
	f.SetCellValue("Sheet1", "A2", 45405)

	date, err := f.GetCellValue("Sheet1", "A2")
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(date)

	if err := f.SaveAs("./Book1.xlsx"); err != nil {
		fmt.Println(err)
	}
}
```

## How Has This Been Tested
Add Unit Test

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your PR. Could you add unit test for this changes?

**Comment by wushiling50** [comment]
> Thanks for your PR. Could you add unit test for this changes?

No problem, it was my oversight, I'll add unit tests for these changes

File: numfmt_test.go [hunk 1]
```diff
@@ -3613,3 +3613,74 @@ func TestNumFmt(t *testing.T) {
 	assert.Equal(t, ErrUnsupportedNumberFormat, err)
 	assert.False(t, changeNumFmtCode)
 }
+
+func TestLangNumFmtFunc(t *testing.T) {
```
Comment (xuri) [inline numfmt_test.go:3617]: Did you test each built-in number format with different data time pattern settings on real Office application?

Comment (wushiling50) [reply numfmt_test.go:3617]: So, what i should do is:
1. Place the numfmt.xlsx file inside the ./test directory.
2. In the tests, set the langNumFmt for the cells in sequence, with each column corresponding to a different region.
3. Retrieve the cell values and compare them to ensure they are correct.

It's not just about the new additions i have made, but also about including zh-cn and en-us as well,right?
In addition to langNumFmt, I also need to do the part of Built-in number format code, right?

Comment (xuri) [reply numfmt_test.go:3617]: You needn't add new workbook to the test directory, the workbook just for test, open the workbook in the real spreadsheet apps to observe formats result in different data time pattern settings, and using the formatted result as expected result in the unit test.

**Comment by wushiling50** [comment]
1. The results of parsing expressions with "e" in Excel and Exelize are different:
demo:  
```go
func main() {
	f := excelize.NewFile()

	numfmt1 := "e\"年\"m\"月\"d\"日\""
	style1, err := f.NewStyle(&excelize.Style{
		CustomNumFmt: &numfmt1,
	})
	if err != nil {
		fmt.Println(err)
		return
	}
	f.SetCellStyle("Sheet1", "A2", "A2", style1)
	f.SetCellValue("Sheet1", "A2", 45050)
	cellValue1, _ := f.GetCellValue("Sheet1", "A2")
	fmt.Printf("1:%v\n", cellValue1)

	numfmt2 := "yyyy\"年\"m\"月\"d\"日\""
	style2, err := f.NewStyle(&excelize.Style{
		CustomNumFmt: &numfmt2,
	})
	if err != nil {
		fmt.Println(err)
		return
	}
	f.SetCellStyle("Sheet1", "B2", "B2", style2)
	f.SetCellValue("Sheet1", "B2", 45050)
	cellValue2, _ := f.GetCellValue("Sheet1", "B2")
	fmt.Printf("2:%v\n", cellValue2)

	numfmt3 := "[$-411]ggge\"年\"m\"月\"d\"日\""
	style3, err := f.NewStyle(&excelize.Style{
		CustomNumFmt: &numfmt3,
	})
	if err != nil {
		fmt.Println(err)
		return
	}
	f.SetCellStyle("Sheet1", "C2", "C2", style3)
	f.SetCellValue("Sheet1", "C2", 45050)

	cellValue3, _ := f.GetCellValue("Sheet1", "C2")
	fmt.Printf("3:%v\n", cellValue3)

	if err := f.SaveAs("demo/Book1.xlsx"); err != nil {
		fmt.Println(err)
	}
}
```
Results in Excel:  
Excel

Results in Exelize:  
Excelize


2. Not supported for Thai language parsing  
demo:  
```go
func main() {
	f := excelize.NewFile()

        numfmt4 := "ว-ดดด-ปป"
	style4, err := f.NewStyle(&excelize.Style{
		CustomNumFmt: &numfmt4,
	})
	if err != nil {
		fmt.Println(err)
		return
	}
	f.SetCellStyle("Sheet1", "D2", "D2", style4)
	f.SetCellValue("Sheet1", "D2", 45050)

	cellValue4, _ := f.GetCellValue("Sheet1", "D2")
	fmt.Printf("4:%v\n", cellValue4)

        if err := f.SaveAs("demo/Book1.xlsx"); err != nil {
		fmt.Println(err)
	}
```
Results in Excel:  
Excel-Thai

Results in Exelize:  
Thai

These problems are leading to the failure of the "zh-tw" and "th-th" related tests, and if you are agreeable, I can provide the code for the "ja-jp" and "ko-kr" parts first, and within the getBuiltInNumFmtCode function, I can set a TODO comment. Both "ja-jp" part and ko-kr" part have passed the tests successfully.

**Comment by xuri** [comment]
Good job. I think we need resolve this number format code parse or evaluate issue before merge this. May be it will be related the NFP library, I'm not sure.

**Comment by xuri** [comment]
Sweet job. I think we need to resolve this number format code parse or evaluate the issue before merging this. Maybe it will be related to the NFP library, I'm not sure.

**Comment by xuri** [comment]
Good job. I think we need to resolve this number format code parse or evaluate the issue before merging this. Maybe it will be related to the NFP library, I'm not sure.

**Review by xuri — APPROVED** [review]

**Review by xuri — COMMENTED** [review]
Thanks for your contribution. I have made some changes based on your branch. This feature will be released in the next version.

**Pull request merged**